apis/r/R/write_seurat.R

```diff
@@ -0,0 +1,539 @@
+#' @importFrom rlang is_integerish
+#'
+NULL
+
+#' Write \pkg{Seurat} Sub-Objects to a SOMA
+#'
+#' Various helpers to write \pkg{Seurat} sub-objects to SOMA
+#'
+#' @inheritParams write_soma_objects
+#' @param x A \pkg{Seurat} sub-object (eg. an
+#' \code{\link[SeuratObject]{Assay}}, a \code{\link[SeuratObject]{DimReduc}},
+#' or a \code{\link[SeuratObject]{Graph}})
+#' @param soma_parent The parent \link[tiledbsoma:SOMACollection]{collection};
+#' for the \code{DimReduc} and \code{Graph} methods, this \strong{must} be a
+#' \link[tiledbsoma:SOMAMeasurement]{measurement} for the assay \code{x}
+#' was generated from
+#'
+#' @name write_soma_seurat_sub_objects
+#' @rdname write_soma_seurat_sub
+#'
+#' @keywords internal
+#'
+NULL
+
+#' @return \code{Assay} method: a \code{\link{SOMAMeasurement}} with the
+#' data from \code{x}
+#'
+#' @rdname write_soma_seurat_sub
+#'
+#' @section Writing \code{\link[SeuratObject]{Assay}s}:
+#' \pkg{Seurat} \code{\link[SeuratObject]{Assay}} objects are written out as
+#' individual \link[tiledbsoma:SOMAMeasurement]{measurements}:
+#' \itemize{
+#'  \item the \dQuote{\code{data}} matrix is written out a
+#'   \link[tiledbsoma:SOMASparseNDArray]{sparse matrix} called
+#'   \dQuote{\code{data}} within the \dQuote{\code{X}} group
+#'  \item the \dQuote{\code{counts}} matrix, if not
+#'   \link[SeuratObject:IsMatrixEmpty]{empty}, is written out a
+#'   \link[tiledbsoma:SOMASparseNDArray]{sparse matrix} called
+#'   \dQuote{\code{counts}} within the \dQuote{\code{X}} group
+#'  \item the \dQuote{\code{scale.data}} matrix, if not
+#'   \link[SeuratObject:IsMatrixEmpty]{empty}, is written out a
+#'   \link[tiledbsoma:SOMASparseNDArray]{sparse matrix} called
+#'   \dQuote{\code{scale_data}} within the \dQuote{\code{X}} group
+#'  \item feature-level meta data is written out as a
+#'   \link[tiledbsoma:SOMADataFrame]{data frame} called \dQuote{\code{var}}
+#' }
+#' Expression matrices are transposed (cells as rows) prior to writing. All
+#' other slots, including results from extended assays (eg. \code{SCTAssay},
+#' \code{ChromatinAssay}) are lost
+#'
+#' @method write_soma Assay
+#' @export
+#'
+write_soma.Assay <- function(
+  x,
+  uri = NULL,
```

Comment:
@aaronwolen @mojaveazure if I understand, this one is OK & necessary -- correct?